atomics/include/desul/atomics/Fetch_Op_Generic.hpp

```diff
@@ -18,38 +18,38 @@ SPDX-License-Identifier: (BSD-3-Clause)
 namespace desul {
 namespace Impl {
 
-#define DESUL_IMPL_ATOMIC_FETCH_OP(ANNOTATION, HOST_OR_DEVICE, OP)        \
-  template <class T, class MemoryOrder, class MemoryScope>                \
-  ANNOTATION T HOST_OR_DEVICE##_atomic_fetch_##OP(                        \
-      T* const dest, const T val, MemoryOrder order, MemoryScope scope) { \
-    return HOST_OR_DEVICE##_atomic_fetch_oper(                            \
-        OP##_operator<T, const T>(), dest, val, order, scope);            \
-  }                                                                       \
-  template <class T, class MemoryOrder, class MemoryScope>                \
-  ANNOTATION T HOST_OR_DEVICE##_atomic_##OP##_fetch(                      \
-      T* const dest, const T val, MemoryOrder order, MemoryScope scope) { \
-    return HOST_OR_DEVICE##_atomic_oper_fetch(                            \
-        OP##_operator<T, const T>(), dest, val, order, scope);            \
+#define DESUL_IMPL_ATOMIC_FETCH_OP(ANNOTATION, HOST_OR_DEVICE, FETCH_OP, OP_U) \
+  template <class T, class MemoryOrder, class MemoryScope>                     \
+  ANNOTATION T HOST_OR_DEVICE##_atomic_##FETCH_OP(                             \
+      T* const dest, const T val, MemoryOrder order, MemoryScope scope) {      \
+    return HOST_OR_DEVICE##_atomic_fetch_oper(                                 \
+        OP_U##operator<T, const T>(), dest, val, order, scope);                \
+  }                                                                            \
+  template <class T, class MemoryOrder, class MemoryScope>                     \
+  ANNOTATION T HOST_OR_DEVICE##_atomic_##OP_U##fetch(                          \
+      T* const dest, const T val, MemoryOrder order, MemoryScope scope) {      \
+    return HOST_OR_DEVICE##_atomic_oper_fetch(                                 \
+        OP_U##operator<T, const T>(), dest, val, order, scope);                \
```

Comment:
This feels like a hack.  I would rename all the foo_operator functors to foo_fetch_operator.